service/api/service.go

```diff
@@ -454,19 +462,23 @@ func (s *serviceImpl) tryPrepareRPCbyDescribe(ctx context.Context, req iwfidl.Wo
 	var searchAttributes []iwfidl.SearchAttribute
 	var dataAttributes []iwfidl.KeyValue
 
-	requestedSAs := req.SearchAttributes
+	var requestedSAs []iwfidl.SearchAttributeKeyAndType
 	saPolicy := req.GetSearchAttributesLoadingPolicy()
-	if saPolicy.GetPersistenceLoadingType() != iwfidl.ALL_WITHOUT_LOCKING {
+	switch saPolicy.GetPersistenceLoadingType() {
+	case iwfidl.PARTIAL_WITH_EXCLUSIVE_LOCK, iwfidl.PARTIAL_WITHOUT_LOCKING:
```

Comment:
it shouldn't run into `PARTIAL_WITH_EXCLUSIVE_LOCK` because it will go to a different path `handleRpcBySynchronousUpdate` if needs locking. 
To be better understand, we should take this case out here -- you can return an error for this